### PR TITLE
feat(ui/menu-select): supports options, labelKey and valueKey

### LINKED ISSUE #1664: menu select component supports options
# Feature request 🚀

### Expected:

- No breaking changes

### Examples:

```js

```

### PR DESCRIPTION
**Pull request by Aybrea**
## Checklist

List of tasks you have already done and plan to do.

- [x] Fix linting errors
- [x] Tests have been added / updated (or snapshots)

## Change information

Supports options usage for component select-menu.

## Issues

close #1664

### CODE REVIEW

File: packages/varlet-ui/src/menu-select/MenuSelectOption.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { defineComponent, type PropType } from 'vue'
```
Comment (chouchouji) [inline packages/varlet-ui/src/menu-select/MenuSelectOption.tsx:1]: 这个文件不需要了

File: packages/varlet-ui/src/menu-select/docs/zh-CN.md [hunk 2]
```diff
@@ -239,6 +304,17 @@ const value = ref()
 | `same-width`    | 是否与触发元素同宽                                                   | _boolean_             | `false`           |
 | `popover-class` | 弹出层的 class                                                    | _string_             | `-`            |
 | `close-on-click-reference` | 是否在点击菜单触发元素后关闭菜单 | _boolean_ | `false` |
+| `options` | 指定可选项 | _SelectOption[]_ | `-` |
```
Comment (chouchouji) [inline packages/varlet-ui/src/menu-select/docs/zh-CN.md:307]: 需要加上版本号

File: packages/varlet-ui/src/menu-select/docs/zh-CN.md [hunk 3]
```diff
@@ -239,6 +304,17 @@ const value = ref()
...
+| `label-key` | 作为 label 唯一标识的键名 | _string_ | `label` |
+| `value-key` | 作为 value 唯一标识的键名 | _string_ | `value` |
+
+#### SelectOption
```
Comment (chouchouji) [inline packages/varlet-ui/src/menu-select/docs/zh-CN.md:311]: MenuSelectOption

File: packages/varlet-ui/src/menu-select/MenuSelect.vue [hunk 4]
```diff
@@ -146,4 +159,5 @@ export default defineComponent({
 @import '../styles/common';
 @import '../menu/menu';
 @import './menuSelect';
+@import '../menu-option/menuOption';
```
Comment (haoziqaq) [inline packages/varlet-ui/src/menu-select/MenuSelect.vue:162]: adjust the order

File: packages/varlet-ui/src/menu-select/MenuSelect.vue [hunk 5]
```diff
@@ -31,6 +31,16 @@
...
         :class="classes(n('menu'), formatElevation(elevation, 3), [scrollable, n('--scrollable')])"
       >
+        <template v-if="menuSelectOptions.length">
+          <var-menu-option
+            v-for="option in menuSelectOptions"
+            :key="option[valueKey]"
+            :label="option[labelKey]"
+            :value="option[valueKey]"
+            :option="option"
+            :disabled="option.disabled"
```
Comment (haoziqaq) [inline packages/varlet-ui/src/menu-select/MenuSelect.vue:41]: set `option.ripple` to `var-menu-option`

File: packages/varlet-ui/src/menu-option/MenuOption.vue [hunk 6]
```diff
@@ -52,21 +52,33 @@ export default defineComponent({
...
     const optionSelected = ref(false)
     const selected = computed(() => optionSelected.value)
-    const label = computed<any>(() => props.label)
+    const labelVNode = computed(() =>
+      isFunction(props.label)
+        ? props.label(
+            props.option ?? {
+              label: props.label,
+              value: props.value,
+              disabled: props.disabled,
```
Comment (haoziqaq) [inline packages/varlet-ui/src/menu-option/MenuOption.vue:69]: support ripple

File: packages/varlet-ui/src/menu-select/docs/en-US.md [hunk 7]
```diff
@@ -213,6 +213,71 @@ const value = ref()
 </template>
 ```
 
+### Option Attribute Usage
```
Comment (haoziqaq) [inline packages/varlet-ui/src/menu-select/docs/en-US.md:216]: Refer to the latest documentation of select component

File: packages/varlet-ui/src/menu-select/docs/en-US.md [hunk 8]
```diff
@@ -247,7 +323,7 @@ const value = ref()
 | `label` | The text that the option displays | _any_ | `-` |
 | `value` | The value of the option binding | _any_ | `-` |
 | `disabled` | Whether to disable | _boolean_ | `false` |
-| `ripple` ***3.3.0***  | Whether to enable ripple | _boolean_ | `true` |
+| `ripple` _**3.3.0**_  | Whether to enable ripple | _boolean_ | `true` |
```
Comment (haoziqaq) [inline packages/varlet-ui/src/menu-select/docs/en-US.md:326]: fix the flag `_`

File: packages/varlet-ui/src/menu-select/docs/en-US.md [hunk 9]
```diff
@@ -239,6 +304,17 @@ const value = ref()
...
 | `close-on-click-reference` | Whether to close the menu when clicking the reference element | _boolean_ | `false` |
+| `options` ***3.3.5*** | Specifies options | _MenuSelectOption[]_ | `-` |
+| `label-key` ***3.3.5*** | As the key that uniquely identifies label | _string_ | `label` |
+| `value-key` ***3.3.5*** | As the key that uniquely identifies value | _string_ | `value` |
+
+#### MenuSelectOption
+
+| Prop | Description | Type | Default |
+| ------- | --- |----------------|-----------|
+| `label`    |   The text of option    | _string \| VNode \| (option: SelectOption, selected: boolean) => VNodeChild_      | `-`   |
```
Comment (chouchouji) [inline packages/varlet-ui/src/menu-select/docs/en-US.md:315]: MenuSelectOption

**Comment by haoziqaq** [comment]
Please bump the version to 3.3.7. thx.

**Comment by Aybrea** [comment]
> Please bump the version to 3.3.7. thx.请将版本提升到 3.3.7。感谢。

done

**Comment by Aybrea** [comment]
> Please bump the version to 3.3.7. thx.

done

**Review by haoziqaq — APPROVED** [review]

**Pull request merged**